src/utils/io/progress_log.rs

```diff
@@ -0,0 +1,238 @@
+        if let Poll::Ready(e) = Pin::new(&mut self.stream).poll_read(cx, buf) {
+            self.progress.inc(buf.filled().len() as u64 - prev_len);
+            Poll::Ready(e)
+        } else {
+            Poll::Pending
+        }
+    }
+}
+
+impl<W: tokio::io::AsyncBufRead + Unpin + tokio::io::AsyncRead> tokio::io::AsyncBufRead
+    for WithProgressStream<W>
+{
+    fn poll_fill_buf(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<io::Result<&[u8]>> {
+        let this = self.get_mut();
+        let result = Pin::new(&mut this.stream).poll_fill_buf(cx);
+        if let Poll::Ready(Ok(buf)) = &result {
+            this.progress.inc(buf.len() as u64);
+        }
+        result
+    }
+
+    fn consume(mut self: Pin<&mut Self>, amt: usize) {
+        Pin::new(&mut self.stream).consume(amt);
+    }
+}
+
+impl<S: futures_core::Stream> futures_core::Stream for WithProgressStream<S> {
+    type Item = S::Item;
+
+    fn poll_next(
+        self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        let this = self.project();
+        let item = this.stream.poll_next(cx);
+        match &item {
+            std::task::Poll::Ready(Some(_)) => {
+                this.progress.inc(1);
+            }
+            std::task::Poll::Ready(None) => this.progress.finish(),
+            std::task::Poll::Pending => {}
+        }
+        item
+    }
+}
+
+#[derive(Debug, Clone)]
+struct WithProgress {
+    completed_items: u64,
+    frequency: Duration,
+    start: Instant,
+    last_logged: Instant,
+    message: String,
+    total_items: u64,
+}
+
+impl WithProgress {
+    fn new(message: &str, total_items: u64) -> Self {
+        let now = Instant::now();
+        Self {
+            completed_items: 0,
+            frequency: UPDATE_FREQUENCY,
+            start: now,
+            last_logged: now,
+            message: message.into(),
+            total_items,
+        }
+    }
+
+    fn inc(&mut self, value: u64) {
+        self.completed_items += value;
+
+        self.emit_log_if_required();
+    }
+
+    fn set(&mut self, value: u64) {
+        self.completed_items = value;
+
+        self.emit_log_if_required();
+    }
+
+    fn set_total(&mut self, value: u64) {
+        self.total_items = value;
+
+        self.emit_log_if_required();
+    }
+
+    fn emit_log_if_required(&mut self) {
+        let now = Instant::now();
+        if (now - self.last_logged) > self.frequency {
+            let elapsed_secs = (now - self.start).as_secs_f64();
+            let elapsed_duration = format_duration(Duration::from_secs(elapsed_secs as u64));
+
+            let throughput = self.completed_items as f64 / elapsed_secs;
+
+            let eta_secs =
+                (self.total_items.saturating_sub(self.completed_items)) as f64 / throughput;
+            let eta_duration = format_duration(Duration::from_secs(eta_secs as u64));
+
+            info!(
```

Comment:
Let's use a different target. Right now, the info messages are tagged with `forest_filecoin::utils::io::progress_log` which is a bit long. Maybe just `forest::progress` would be enough.